PilotDataPlatform/metadata
# PILOT-1484 Implement PATCH endpoint for pinning/unpinning favourites

## Summary

- Add `PATCH /v1/favourites` endpoint for pinning/unpinning favourite items/collections.

## JIRA Issues

https://indocconsortium.atlassian.net/browse/PILOT-1484

## Type of Change


## Testing

Are there any new or updated tests to validate the changes?


## Test Directions

1. Manually create some favourites.
2. Pin/unpin those favourites via the new endpoint and check that the responses are correct.
3. Review the [API design](https://indocconsortium.atlassian.net/wiki/spaces/PILOT/pages/3189309472/Favourites+API+design) and ensure that the implementation matches the design.
